src/negotiation/order.ts

```diff
@@ -16,8 +16,125 @@ export interface OrderAsset {
   nominalAmount: string;
 }
 
+export interface TakerCriteria {
+  buy: TakerCriteriaAsset;
+  sell: TakerCriteriaAsset;
+  minRate?: number;
+}
+
+export interface TakerCriteriaAsset {
+  ledger: string;
+  asset: string;
+  minNominalAmount?: string;
+  maxNominalAmount?: string;
+}
+
+/**
+ * @description An order handler for the taker. It has helper functions to facilitate the handler of an
```

Comment:
🔧 `handler` for me implies, that this class **deals with** orders rather than **being** an `Order`.

Reply:
I rephrased.